core/domain/blog_domain.py

```diff
@@ -476,18 +476,19 @@ def validate(self, strict: bool = False) -> None:
                     'Expected thumbnail filename to be a string, received: %s.'
                     % self.thumbnail_filename
                 )
+
             self.require_valid_thumbnail_filename(
                 self.thumbnail_filename, strict=strict)
 
-        if not isinstance(self.summary, str):
-            raise utils.ValidationError(
-                'Expected summary to be a string, received: %s' % self.summary)
-
-        if strict:
             self.require_valid_url_fragment(self.url_fragment)
+
             if not self.summary:
                 raise utils.ValidationError('Summary can not be empty')
 
+        if not isinstance(self.summary, str):
+            raise utils.ValidationError(
+                'Expected summary to be a string, received: %s' % self.summary)
+
```

Comment:
Why is this moved?

Reply:
Ditto as above. Combined 2 separate `if strict` blocks into one!